src/sentry/snuba/discover.py

```diff
@@ -280,6 +280,7 @@ def query(
     reference_event=None,
     referrer=None,
     auto_fields=False,
+    ignore_aggregate_conditions=False,
```

Comment:
Would we be able to make this an implicit behaviour of the discover wrappers? We know both the fields and the having conditions. We could elide any having clauses we know will be invalid, or would that conflict with the scenarios we want to let users know about their mistakes?

Reply:
The reason this check is here is at your request, since if we just remove invalid having conditions, the user will get back unexpected results.

Reply:
Makes sense.